nodeup/pkg/model/protokube.go

```diff
@@ -332,11 +333,15 @@ func (t *ProtokubeBuilder) ProtokubeFlags(k8sVersion semver.Version) (*Protokube
 		f.DNSInternalSuffix = fi.String(internalSuffix)
 	}
 
-	if t.Cluster.Spec.CloudProvider != "" {
-		f.Cloud = fi.String(t.Cluster.Spec.CloudProvider)
+	if cloudProvider := t.Cluster.Spec.CloudProvider; cloudProvider != "" {
+		if kops.CloudProviderID(cloudProvider) == kops.CloudProviderSpotinst {
+			cloudProvider = string(spotinst.GuessCloudFromClusterSpec(&t.Cluster.Spec))
```

Comment:
What are the other options this could return? If it's always going to be AWS for spotinst: Maybe it's easier to tag spotinst with an additional flag and keep the cloudProvider set as AWS.

Reply:
It's going to be AWS for Spotinst for now, but we are working on adding support for additional cloud providers (GCE, ALI, etc.).

Reply:
In that regard I’d propose even more to add something like a flag that it’s managed via spotinst in favor of adding separate providers and remapping them to the ones we already have. 

Otherwise we will build edge case after edge case here and the logic will get exponentially more complex per spotinst provider you add

Reply:
My assumption here is - correct me if I am wrong - that the cluster configuration for spotinst AWS is mainly the same, just extended by some fields. 

If we now implement it the way you did that results in the following: 
Every cloud provider you add with spotinst will be an alternative case for the given cloud vendor - resulting in lot's of changes all across kops. 

I would personally treat spotinst more like a switch that - if the flag "this is for spotinst, not aws, gcloud, e.g. directly - would take over, replace the few parts that require modification and process in protokube. 

Keeps our processing clean and this PR would drastically shrink in size I assume.

Reply:
See [comment](https://github.com/kubernetes/kops/pull/5195#discussion_r204143415) above.